packages/suite/src/middlewares/wallet/__fixtures__/coinjoinMiddleware.ts

```diff
@@ -0,0 +1,287 @@
+            accountKey: 'account-B-key',
+            isStarting: true,
+        },
+    },
+    {
+        type: COINJOIN.SESSION_RESTORE,
+        payload: {
+            accountKey: 'account-B-key',
+        },
+    },
+    {
+        type: COINJOIN.SESSION_STARTING,
+        payload: {
+            accountKey: 'account-B-key',
+            isStarting: false,
+        },
+    },
+];
+
+export const fixtures = [
+    {
+        description: 'pause coinjoin session when remembered device disconnects',
+        state: DEFAULT_STATE,
+        action: {
+            type: DEVICE.DISCONNECT,
+            payload: {
+                id: 'device-A-id',
+            },
+        },
+        expectedActions: [
+            {
+                type: COINJOIN.SESSION_PAUSE,
+                payload: {
+                    accountKey: 'account-A-key',
+                    interrupted: false,
+                },
+            },
+        ],
+    },
+    {
+        description: 'interrupt all coinjoin sessions when Tor is disabled',
+        state: DEFAULT_STATE,
+        action: {
+            type: SUITE.TOR_STATUS,
+            payload: 'Disabled',
+        },
+        expectedActions: PAUSE_ALL_INTERRUPTED_SESSIONS_ACTIONS,
+    },
+    {
+        description: 'restore all interrupted coinjoin sessions when Tor is enabled',
+        state: STATE_WITH_INTERRUPTED_SESSION,
+        client: 'btc' as NetworkSymbol,
+        connect: {
+            success: true,
+        },
+        action: {
+            type: SUITE.TOR_STATUS,
+            payload: 'Enabled',
+        },
+        expectedActions: RESTORE_SESSION_B_ACTIONS,
+    },
+    {
+        description: 'interrupt current coinjoin session when user enters send form',
+        state: DEFAULT_STATE,
+        action: {
+            type: ROUTER.LOCATION_CHANGE,
+            payload: {
+                route: {
+                    name: 'wallet-send',
+                },
+            },
+        },
+        expectedActions: [
+            {
+                type: COINJOIN.SESSION_PAUSE,
+                payload: {
+                    accountKey: 'account-B-key',
+                    interrupted: true,
+                },
+            },
+        ],
+    },
+    {
+        description: 'restore all interrupted coinjoin sessions when user leaves send form',
+        state: STATE_WITH_INTERRUPTED_SESSION,
+        client: 'btc' as NetworkSymbol,
+        connect: [
+            {
+                success: true,
+            },
+        ],
+        action: {
+            type: ROUTER.LOCATION_CHANGE,
+            payload: {
+                route: {
+                    name: 'settings-index',
+                },
+                settingsBackRoute: {
+                    name: 'wallet-send',
+                },
+            },
+        },
+        expectedActions: RESTORE_SESSION_B_ACTIONS,
+    },
+    {
+        description: 'interrupt related coinjoin session when an account goes out of sync',
```

Comment:
check that it does not interrupt sessions in critical phases?

Reply:
0dee27a7e4512783a15c46301b1be41104f2e163